### ISSUE TITLE
add grunt yui_dev task *or* add grunt yui_production task, rather than using env var

### ISSUE THREAD

**Issue by toolness** [processing/p5.js#1129]
So one of the downsides to #1124 is that by default, the `<script>` links to the p5 libraries are actually set to point to their built locations on the *development* server, e.g. `/lib/p5.js` and `lib/addons/p5.dom.js`. This can be overridden to point to the libraries on p5js.org if the deployer runs `export P5_SITE_ROOT=..` before running `grunt`.

Because (I think?) this wasn't done when building the reference docs for deployment to the real p5js.org, those script links are now broken, which means most of the reference on p5js.org is broken right now. :disappointed: 

I feel conflicted about my solution to use `P5_SITE_ROOT` since it's so easy to forget to set it for production.

One possible solution is to actually make `/lib/p5.js` and the others not 404 on p5js.org, either through symlinks or by just moving those files... Any other ideas?

**Comment by lmccart** [comment]
ahh oops. i can rebuild.
could we change it so that the default builds properly for the site?
or could we add another task like `grunt yui_test` that builds them using a different path? maybe this is the best because people don't have to figure out how to set the env variable, or what they should be set too. is it possible to do this with the way the grunt tasks and docs build are currently configured? I haven't looked too closely yet...

**Comment by toolness** [comment]
Yeah, I agree that adding a new `grunt` command is probably better than an environment variable. 

When I added the current functionality, my philosophy was that any configuration defaults should favor the newbie developer experience, which is why I made the default build properly for development mode rather than production... I can reverse that though if you want.

What do you think of `grunt yui_production` building the docs for production? Or should `grunt yui` build for production and something like `grunt yui_dev` build development docs?

Another alternative is to build *both* dev and production docs at the same time, and put them in separate directories... That could result in a more complicated grunt file and slow down build times though.

**Comment by lmccart** [comment]
I see the point of favoring newbie devs, but I guess my worry is that people that are not exactly newbies but haven't read closely may try to generate the docs and submit to the website, and it seems likely that we may end up with broken docs from time to time. so I think there are pros and cons to favoring either, I'm not sure which `grunt yui` should default to, but I like the idea of either `grunt yui_production` or `grunt yui_dev`, it seems easier to switch between the two than setting an env var. I'd say go with whatever your intuition tells you in terms of naming, and let's just make sure both are listed clearly in the dev docs.

**Comment by lmccart** [comment]
I set it up so `grunt yui` will production ready docs and `grunt yui:dev` will build the dev docs linking to local copies of the lib. However, I just duplicated all the yuidoc options, I'm not sure if there's a DRYer way to do it where we only update the helpers line. I tried some things but couldn't get it to work. @toolness let me know if you have any thoughts?
https://github.com/processing/p5.js/blob/master/Gruntfile.js#L225
